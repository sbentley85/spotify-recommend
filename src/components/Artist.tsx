import { url } from "inspector";
import React from "react";
import { IPicks } from "./Choices";

const Artist = (props: { artist: IPicks; handlePicks?: any }) => {
	return (
		<div
			onClick={(e) => props.handlePicks(e, props.artist)}
			className="artist"
		>
			<div
				className="artistImage"
				style={
					props.artist.smImg
						? { backgroundImage: `url(${props.artist.smImg.url})` }
						: { backgroundColor: "black" }
				}
			></div>
			<div className="artistName">{props.artist.name}</div>
		</div>
	);
};

export default Artist;
