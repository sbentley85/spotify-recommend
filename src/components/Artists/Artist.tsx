import React from "react";
import { IPicks, IPickCallback } from "../Choices/Choices";
import artistStyles from "../Tracks/tracks.module.css";

const Artist = (props: { artist: IPicks; handleClick: IPickCallback }) => {
	return (
		<div
			onClick={(e) => props.handleClick(e, props.artist)}
			className={artistStyles.artist}
		>
			<div
				className={artistStyles.artistImage}
				style={
					props.artist.smImg
						? { backgroundImage: `url(${props.artist.smImg.url})` }
						: { backgroundColor: "black" }
				}
			></div>
			<div className={artistStyles.artistName}>{props.artist.name}</div>
		</div>
	);
};

export default Artist;
