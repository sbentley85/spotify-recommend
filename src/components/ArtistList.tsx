import React from 'react';
import Artist from './Artist'

const ArtistList = (props: {artists: {name: string, id: string}[]}) => {
	return (
		<div>
			{props.artists.map((artist: {name: string, id: string}) => {
				return <Artist artist={artist} key={artist.id}/>
			})}
		</div>
	);
};

export default ArtistList;