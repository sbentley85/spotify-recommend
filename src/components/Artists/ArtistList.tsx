import React from "react";
import Artist from "./Artist";
import { IPicks } from "../Choices/Choices";

const ArtistList = (props: { artists: IPicks[]; handleClick?: any }) => {
	return props.artists.length !== 0 ? (
		<div className="artistList">
			{props.artists.map((artist: IPicks) => {
				return (
					<Artist
						handleClick={props.handleClick}
						artist={artist}
						key={artist.id}
					/>
				);
			})}
		</div>
	) : null;
};

export default ArtistList;
