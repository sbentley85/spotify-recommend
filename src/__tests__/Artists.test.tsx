import React from "react";
import Artist from "../components/Artists/Artist";
import { shallow } from "enzyme";
import ArtistList from "../components/Artists/ArtistList";

// sample track
const sampleArtist = {
	id: "3CjlHNtplJyTf9npxaPl5w",
	name: "CHVRCHES",
	medImg: {
		height: 320,
		url: "https://i.scdn.co/image/64a6c54b7bea1c2c314da545c3f5048d28f48c3b",
		width: 320,
	},
	smImg: {
		height: 160,
		url: "https://i.scdn.co/image/a5b058c5287ade15994f668a46f5d7f782dae97e",
		width: 160,
	},
	uri: "spotify:artist:3CjlHNtplJyTf9npxaPl5w",
};

const sampleArtists = [
	{
		id: "3CjlHNtplJyTf9npxaPl5w",
		name: "CHVRCHES",
		medImg: {
			height: 320,
			url:
				"https://i.scdn.co/image/64a6c54b7bea1c2c314da545c3f5048d28f48c3b",
			width: 320,
		},
		smImg: {
			height: 160,
			url:
				"https://i.scdn.co/image/a5b058c5287ade15994f668a46f5d7f782dae97e",
			width: 160,
		},
		uri: "spotify:artist:3CjlHNtplJyTf9npxaPl5w",
	},
	{
		id: "3TVXtAsR1Inumwj472S9r4",
		name: "Drake",
		medImg: {
			height: 320,
			url:
				"https://i.scdn.co/image/5ea794cf832550943d5f8122afcf5f23ee9d85b7",
			width: 320,
		},
		smImg: {
			height: 160,
			url:
				"https://i.scdn.co/image/8eaace74aaca82eaccde400bbcab2653b9cf86e1",
			width: 160,
		},
		uri: "spotify:artist:3TVXtAsR1Inumwj472S9r4",
	},
];

describe("Artist Component", () => {
	it("renders Artist component", () => {
		shallow(
			<Artist
				artist={sampleArtist}
				handleClick={() => {
					console.log("testing Artist onClick");
				}}
			/>
		);
	});

	it("renders Artist details from props", () => {
		const myArtist = shallow(
			<Artist
				artist={sampleArtist}
				handleClick={() => {
					console.log("testing Artist onClick");
				}}
			/>
		);
		expect(myArtist).toMatchSnapshot();
	});

	const clickFn = jest.fn();

	it("Artist click should call onClick function", () => {
		const myArtist = shallow(
			<Artist artist={sampleArtist} handleClick={clickFn} />
		);
		myArtist.simulate("click");
		expect(clickFn).toHaveBeenCalled();
	});
});

describe("ArtistList component", () => {
	const clickFn = jest.fn();
	it("renders ArtistList without crashing", () => {
		shallow(<ArtistList artists={sampleArtists} handleClick={clickFn} />);
	});
	it("renders a list of Artist components from props", () => {
		const myArtistList = shallow(
			<ArtistList artists={sampleArtists} handleClick={clickFn} />
		);
		expect(myArtistList).toMatchSnapshot();
	});
});
