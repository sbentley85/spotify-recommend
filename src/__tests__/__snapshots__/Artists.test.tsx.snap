// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist Component renders Artist details from props 1`] = `
<div
  className="artist"
  onClick={[Function]}
>
  <div
    className="artistImage"
    style={
      Object {
        "backgroundImage": "url(https://i.scdn.co/image/a5b058c5287ade15994f668a46f5d7f782dae97e)",
      }
    }
  />
  <div
    className="artistName"
  >
    CHVRCHES
  </div>
</div>
`;

exports[`ArtistList component renders a list of Artist components from props 1`] = `
<div
  className="artistList"
>
  <Artist
    artist={
      Object {
        "id": "3CjlHNtplJyTf9npxaPl5w",
        "medImg": Object {
          "height": 320,
          "url": "https://i.scdn.co/image/64a6c54b7bea1c2c314da545c3f5048d28f48c3b",
          "width": 320,
        },
        "name": "CHVRCHES",
        "smImg": Object {
          "height": 160,
          "url": "https://i.scdn.co/image/a5b058c5287ade15994f668a46f5d7f782dae97e",
          "width": 160,
        },
        "uri": "spotify:artist:3CjlHNtplJyTf9npxaPl5w",
      }
    }
    handleClick={[MockFunction]}
    key="3CjlHNtplJyTf9npxaPl5w"
  />
  <Artist
    artist={
      Object {
        "id": "3TVXtAsR1Inumwj472S9r4",
        "medImg": Object {
          "height": 320,
          "url": "https://i.scdn.co/image/5ea794cf832550943d5f8122afcf5f23ee9d85b7",
          "width": 320,
        },
        "name": "Drake",
        "smImg": Object {
          "height": 160,
          "url": "https://i.scdn.co/image/8eaace74aaca82eaccde400bbcab2653b9cf86e1",
          "width": 160,
        },
        "uri": "spotify:artist:3TVXtAsR1Inumwj472S9r4",
      }
    }
    handleClick={[MockFunction]}
    key="3TVXtAsR1Inumwj472S9r4"
  />
</div>
`;
